lib/src/Uncertainty/Algorithm/MetaModel/MetaModelAlgorithm.cxx

```diff
@@ -109,4 +115,75 @@ void MetaModelAlgorithm::load(Advocate & adv)
   adv.loadAttribute( "model_", model_ );
 }
 
+
+Distribution MetaModelAlgorithm::BuildDistribution(const Sample & inputSample)
+{
+  // Recover the distribution, taking into account that we look for performance
+  // so we avoid to rebuild expensive distributions as much as possible
+  const UnsignedInteger inputDimension = inputSample.getDimension();
+  Collection< Distribution > marginals(inputDimension);
+
+  // The strategy for the marginals is to find the best continuous 1-d parametric model
+  Collection< DistributionFactory > factories(DistributionFactory::GetContinuousUniVariateFactories());
+
+  // exclude non-parametric or costly ones
+  const Description excluded = {"HistogramFactory", "TrapezoidalFactory", "FrechetFactory", "MeixnerDistributionFactory", "TruncatedNormalFactory"};
```

Comment:
(I fear that my comment gets lost, so I post it twice).
Using the BIC as a default corresponds exactly to one of my wishes. However, this particular code with the "excluded" variable does not correspond to what I would suggest. I understand that we want to exclude the Histogram (because it always fit very well) and the Trapezoidal, Frechet, Meixner and Truncated, because fitting them costs a lot.
* Since the BuildDistribution searches for a Distribution which fits the sample, it might be clearer if the list of factories is an explicit input argument of BuildDistribution. This would clarify that this list is used to select the distribution which best matches the sample. Adding a list of copula might also be necessary.
* I would rather create a new GetContinuousPolynomialChaosUniVariateFactories method in order to include, rather than exclude a hard-coded list. This would allow to motivate the list of distribution that we include within the help of this particular method. Moreover, adding a new exotic distribution in the lib would not force to include it, while the current code would force to exclude it (if we do not simply forget this particular point).
* If the code is unchanged in its structure, the minimum would be to add the FisherSnedecor, because, according to #1061, this costs a lot. Moreover, I would create a new entry in the ResourceMap so that we can customize the list.